parquet-variant/src/main/java/org/apache/parquet/variant/VariantArrayBuilder.java

```diff
@@ -28,10 +28,15 @@ public class VariantArrayBuilder extends VariantBuilder {
   protected long numValues = 0;
 
   VariantArrayBuilder(VariantBuilder rootBuilder) {
+    super(rootBuilder.metadata);
```

Comment:
This is needed to use the same metadata when building nested child objects. Same with the object builder.